File: 프로그래머스/0/120875. 평행/평행.java
class Solution {
    public int solution(int[][] dots) {
        return isParallel(dots[0], dots[1], dots[2], dots[3]) ||
               isParallel(dots[0], dots[2], dots[1], dots[3]) ||
               isParallel(dots[0], dots[3], dots[1], dots[2]) ? 1 : 0;
    }
    
    private boolean isParallel(int[] a, int[] b, int[] c, int[] d) {
        int dx1 = b[0] - a[0];
        int dy1 = b[1] - a[1];
        int dx2 = d[0] - c[0];
        int dy2 = d[1] - c[1];
        return dy1 * dx2 == dy2 * dx1;
    }
}

File: 프로그래머스/0/120875. 평행/평행.py
def solution(dots):
    # 점들을 변수에 할당 (가독성을 위해)
    a, b, c, d = dots
    
    # 3가지 경우의 수 체크
    # (a-b, c-d), (a-c, b-d), (a-d, b-c)
    if (a[1]-b[1])*(c[0]-d[0]) == (c[1]-d[1])*(a[0]-b[0]): return 1
    if (a[1]-c[1])*(b[0]-d[0]) == (b[1]-d[1])*(a[0]-c[0]): return 1
    if (a[1]-d[1])*(b[0]-c[0]) == (b[1]-c[1])*(a[0]-d[0]): return 1
    
    return 0